app/src/main/java/org/schabi/newpipe/App.java

```diff
@@ -100,7 +104,13 @@ public void onCreate() {
         configureRxJavaErrorHandler();
 
         // Check for new version
-        new CheckForNewAppVersionTask().execute();
+        disposable = CheckForNewAppVersion.checkNewVersion(App.getApp());
+    }
+
+    @Override
+    public void onTerminate() {
+        disposable.dispose();
```

Comment:
Make `private Disposable disposable = null` and add a null check here, just to be sure.